README.md

```diff
@@ -129,6 +129,92 @@ The image ships with some basic fonts (`font-terminus font-inconsolata font-deja
 - Alpine Linux see https://wiki.alpinelinux.org/wiki/Fonts
 - Debian/Ubuntu see https://wiki.debian.org/Fonts
 
+#### Custom fonts directory
+
+Let's say your building a docker image for you application, where you want to use the `ffmpeg` and `ffprobe` binaries, and  want to have a dedicated directory for storing fonts, e.g. `/app/fonts`. In order to achieve this, let's take a look at the following steps.
+
+1. The library `fontconfig` has to be installed in your image and the cache directory has to be writable (for non-root users).
```

Comment:
I wonder if it's not strictly needed to install a fontconfig package from the distro used. I tried to skip ut and did this:
```Dockerfile
FROM python:3.11-slim-bookworm AS base

COPY <<EOF /etc/fonts/conf.d/50-custom.conf
<?xml version="1.0"?>
<!DOCTYPE fontconfig SYSTEM "urn:fontconfig:fonts.dtd">
<fontconfig>
        <dir>/app/fonts</dir>
</fontconfig>
EOF

FROM base as ffmpeg

COPY --from=mwader/static-ffmpeg:7.0.1 /ffmpeg /usr/bin/
COPY --from=mwader/static-ffmpeg:7.0.1 /ffprobe /usr/bin/

FROM ffmpeg AS app

WORKDIR /app

RUN <<EOT bash
  set -ex
  mkdir -p ./{fonts}
EOT
```

```sh
$ docker build -t test .
...
$ ls -l fonts
total 392
-rw-r--r--@ 1 wader  staff  197584 Dec 14  2023 Ubuntu Mono derivative Powerline.ttf

$ docker run --rm -w $PWD -v $PWD:$PWD -v $PWD/fonts:/app/fonts -v $PWD/cache:/var/cache/fontconfig --entrypoint=/usr/bin/ffmpeg test -v debug -y -f lavfi -i 'color=white,drawtext=text=bla:fontfile=Ubuntu Mono derivative Powerline' -t 1s bla.mp4

$ ls -l cache
total 8
-rw-r--r--  1 wader  staff  2600 Jun 28 22:06 12bd695e3d0fbb7a6dd24a23468f757a-le64.cache-9

```

But i did notice that when i installed the `fontconfig` package there will be more cache files, maybe because there will be additional fontconfig files in `/etc/fonts/conf.d` so more fonts gets loaded/cached?

Reply:
> I wonder if it's not strictly needed to install a fontconfig package from the distro used. 

> ```shell
> $ docker run --rm -w $PWD -v $PWD:$PWD -v $PWD/fonts:/app/fonts -v $PWD/cache:/var/cache/fontconfig --entrypoint=/usr/bin/ffmpeg test -v debug -y -f lavfi -i 'color=white,drawtext=text=bla:fontfile=Ubuntu Mono derivative Powerline' -t 1s bla.mp4
> ```

Hmm interesting, seems you're right about that. Doesn't matter root/non-root user either. It seems that without `fontconfig` addionally installed, when you mount `$PWD/cache:/var/cache/fontconfig`, it will write cache to that directory. And if you omit the mount then it will never write cache to `/var/cache/fontconfig`, because the directory doesn't exist in the base image (which in hindsight makes sense).

Reply:
Yeap i think it would be ideal if things could work without installing additional program etc, and it seems so? so then it should just be to copy some files and from the static-ffmpeg image for standard config and fonts and/or volume mount some directory or file for optional cache and to add additional font files? is that what most ppl would like to know how to do?

BTW we could also link to https://fontconfig.pages.freedesktop.org/fontconfig/fontconfig-user.html

Reply:
> Yeap i think it would be ideal if things could work without installing additional program etc, and it seems so?

Agreed. I'm just going to test this again in couple of my images to see if it works like intended without actually needing to install `fontconfig`.

> so then it should just be to copy some files and from the static-ffmpeg image for standard config and fonts and/or volume mount some directory or file for optional cache and to add additional font files? is that what most ppl would like to know how to do?

I think showing to mount fonts (and optionally cache directory) is interesting, basically the example you've already showed previously.

```shell
docker run --rm -w $PWD -v $PWD:$PWD -v $PWD/fonts:/app/fonts -v $PWD/cache:/var/cache/fontconfig --entrypoint=/usr/bin/ffmpeg test -v debug -y -f lavfi -i 'color=white,drawtext=text=bla:fontfile=Ubuntu Mono derivative Powerline' -t 1s bla.mp4
```

> BTW we could also link to https://fontconfig.pages.freedesktop.org/fontconfig/fontconfig-user.html

Good idea.